### PR TITLE
Studies Hub: Tab Navigator

### PR DESCRIPTION
**Pull request by juliatan**
# Description

Notion: https://www.notion.so/joinzoe/Studies-hub-Tabbed-nav-f137985f50ba4c91b949929d350c252c

Note:
- Need to complete testing suite

## On what platform have you tested the change?

- [x] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Out of scope and potential follow-up

[comment]: <> (Are there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?)

## Screenshots

| Android | iOS |
| - | - |
| Screenshot_1635520079 | Simulator Screen Shot - iPhone 13 - 2021-10-29 at 16 10 17 |

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if relevant)
- [ ] Added new screen(s) to `src/features/screens.ts` and new Redux slice(s) to `mocks/mockedInitialState.ts` (if relevant)

## Checklist (for reviewers)

- [x] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Checklist (for merger)

- [ ] Make sure the right merge strategy is chosen, commit merge into master and squash merge into develop.

### CODE REVIEW

**Comment by selvach** [comment]
🥇

File: assets/icons/navigation/HomeIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { TSvgProps } from '@assets/Svg';
```
Comment (aikewoody) [inline assets/icons/navigation/HomeIcon.tsx:1]: I am still of the opinion that this doesn't belong in the assets directory and should be put in the components directory of the feature

File: src/features/tab-navigation/TabNavigator.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+export default function TabNavigator() {
+  return (
+    <Tab.Navigator
+      initialRouteName="Home"
+      screenOptions={{
+        ...tabNavigatorScreenOptions,
+        ...(Platform.OS === 'android' ? tabNavigatorScreenOptionsAndroidOnly : null),
+      }}
+    >
+      <Tab.Screen component={HomeScreen} name="Home" options={{ ...tabScreenOptions, ...tabHomeScreenOptions }} />
```
Comment (aikewoody) [inline src/features/tab-navigation/TabNavigator.tsx:98]: As these options are static data they can be defined outside the component function which results in more efficient code

File: src/features/tab-navigation/TabNavigator.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+      <Tab.Screen
+        component={StudiesListScreen}
+        name="Studies"
+        options={{ ...tabScreenOptions, ...tabStudiesScreenOptions }}
```
Comment (aikewoody) [inline src/features/tab-navigation/TabNavigator.tsx:102]: Same for these options

File: src/features/tab-navigation/__tests__/TabNavigator.test.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  });
+
+  it('shows Home screen stack when selected', async () => {
+    const instance = createComponent().instance;
+    expect(instance.findAllByType(MainNavigator).length).toBe(1);
+    // add expectation to not see the studies stack
+  });
+
+  // TODO: add tests to show clicking on studies tab works as expected
+});
```
Comment (aikewoody) [inline src/features/tab-navigation/__tests__/TabNavigator.test.tsx:55]: Great how you always create tests for new features!

**Pull request closed without merging**